include/deal.II/fe/fe_values.h

```diff
@@ -292,6 +316,13 @@ namespace FEValuesViews
                               std::vector<typename ProductType<value_type,typename InputVector::value_type>::type> &values) const;
 
     /**
+     *
+     */
+    template <class InputVector>
+    void get_function_values_from_local_dof_values (const InputVector &dof_values,
+                                                    std::vector<typename OutputType<typename InputVector::value_type>::value_type> &values) const;
```

Comment:
I agree. The problematic case that we're working towards is this: some Sacado number types generate a new class types for each operation, so the return type when `Number` is of Sacado (or some other AD types) is actually non-trivial, i.e.
```
ProductType<double,Sacado::Fad::DFad<double>>::type != {double, Sacado::Fad::DFad<double>}!
```